content/en/spec/oauth.md

```diff
@@ -13,41 +13,76 @@ The Mastodon API has many methods that require authentication from a client or a
 
 {{< caption-link url="https://github.com/mastodon/mastodon/blob/main/config/initializers/doorkeeper.rb" caption="Doorkeeper config initializer" >}}
 
-### Authorization endpoint (RFC 6749 Section 3.1) {#authorization}
+### Authorization Server Metadata endpoint ([RFC 8414](https://www.rfc-editor.org/rfc/rfc8414.html)) {#authorization-server-metadata}
+
+Returns a JSON document representing the configuration of the OAuth 2 server in Mastodon. Information includes `scopes` available for use when [registering Applications]({{% relref "methods/apps#create" %}}) or requesting [access tokens]({{% relref "methods/oauth#token" %}}), `grant_types_supported` which are can be used when requesting access tokens, and various endpoints for interacting with the Mastodon OAuth server, such as `authorization_endpoint` and `token_endpoint`.
+
+**Version history:**\
+4.3.0 - added
+
+{{< page-relref ref="methods/oauth#authorization-server-metadata" caption="GET /.well-known/oauth-authorization-server" >}}
 
-[GET /oauth/authorize]({{% relref "methods/oauth#authorize" %}})
+### Dynamic Client Registration endpoint ([RFC 7591](https://www.rfc-editor.org/rfc/rfc7591.html)) {#dynamic-client-registration}
+
+At present, Mastodon does not support the Dynamic Client Registration protocol, however it does support a proprietary endpoint for registering an OAuth Application.
+
+{{< page-relref ref="methods/apps#create" caption="POST /api/v1/apps" >}}
+
+### Authorization endpoint ([RFC 6749 Section 3.1](https://www.rfc-editor.org/rfc/rfc6749.html#section-3.1)) {#authorization}
 
 Displays an authorization form to the user. If approved, it will create and return an authorization code, then redirect to the desired `redirect_uri`, or show the authorization code if `urn:ietf:wg:oauth:2.0:oob` was requested.
 
-### Token endpoint (RFC 6749 Section 3.2) {#token}
+{{< page-relref ref="methods/oauth#authorize" caption="GET /oauth/authorize" >}}
 
-[POST /oauth/token]({{% relref "methods/oauth#token" %}})
+### Token endpoint ([RFC 6749 Section 3.2](https://www.rfc-editor.org/rfc/rfc6749.html#section-3.2)) {#token}
 
 Obtain an access token. Mastodon supports the following OAuth 2 flows:
 
 Authorization code flow
 : For end-users
 
-Password grant flow
-: For bots and other single-user applications
-
 Client credentials flow
 : For applications that do not act on behalf of users
 
-### Token revocation endpoint (RFC 7009 Section 2) {#revoke}
+Mastodon has historically supported the Password Grant flow, however, usage is [not recommended](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics#name-resource-owner-password-cre) by the OAuth 2 Specification authors due to security issues, and has subsequently been removed from future versions of Mastodon. Instead, it is recommended that you create an OAuth Application for that user, and use the generated Access Token for interacting with the API.
```

Comment:
@ClearlyClaire This is the change I did for the removal of the Password Grant Flow Type. Since it's not actually deprecated in 4.3, I've just made it a paragraph explaining it has been supported, but not when it's removed.